test/e2e/service.go

```diff
@@ -464,8 +464,50 @@ var _ = ginkgo.Describe("Services", func() {
 				})
 			})
 		})
+
 	}
 
+	ginkgo.It("does not use host masquerade address as source IP address when communicating externally", func() {
```

Comment:
would this test case normally fail without your deprecation fix? If so, why? I would have expected to see some modification to the host to get into a situation where it would attempt to pick the masquerade IP, like how the original bug talks about noprefixroute or whatever.

Reply:
yes, it will fail when I don't apply my fix.

The reason is there is no route and subnet that includes the dest addr and the default route for IPv6 doesnt specify a src IP to use unlike the V4 setup.